src/node_http2.cc

```diff
@@ -349,11 +348,11 @@ Http2Priority::Http2Priority(Environment* env,
   Local<Context> context = env->context();
   int32_t parent_ = parent->Int32Value(context).ToChecked();
   int32_t weight_ = weight->Int32Value(context).ToChecked();
-  bool exclusive_ = exclusive->BooleanValue(env->isolate());
+  bool exclusive_ = exclusive->IsTrue();
   Debug(env, DebugCategory::HTTP2STREAM,
-        "Http2Priority: parent: %d, weight: %d, exclusive: %d\n",
-        parent_, weight_, exclusive_);
-  nghttp2_priority_spec_init(&spec, parent_, weight_, exclusive_ ? 1 : 0);
+        "Http2Priority: parent: %d, weight: %d, exclusive: %s\n",
+        parent_, weight_, exclusive_ ? "yes" : "no");
```

Comment:
Fwiw, `Debug()`/`SPrintF()` handles booleans well (unlike raw `fprintf`), this should not be necessary

Reply:
It was triggering a warning when compiling on Windows.

```
C:\Users\jasne\Projects\node\src\node_http2.cc(2738,13): warning C4244: 'initializing': conversion from 'T' to 'int', possible loss of data [C:\Users\jasne\Projects\node\libnode.vcxproj]
          with
          [
              T=int64_t
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(39,1): warning C4806: '&': unsafe operation: no value of type 'T' promoted to type 'int' can equal the given constant [C:\Users\jasne\Projects\node\libnode.vcxproj]
          with
          [
              T=bool
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(60): message : see reference to function template instantiation 'std::string node::ToStringHelper::BaseConvert<3,bool,0>(T)' being compiled [C:\Users\jasne\Projects\node\lib
node.vcxproj]
          with
          [
              T=bool
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(97): message : see reference to function template instantiation 'std::string node::ToBaseString<3,bool&>(T)' being compiled [C:\Users\jasne\Projects\node\libnode.vcxproj]
          with
          [
              T=bool &
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(117): message : see reference to function template instantiation 'std::string node::SPrintFImpl<bool&,>(const char *,Arg)' being compiled [C:\Users\jasne\Projects\node\libno
de.vcxproj]
          with
          [
              Arg=bool &
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(117): message : see reference to function template instantiation 'std::string node::SPrintFImpl<int&,bool&>(const char *,Arg,bool &)' being compiled [C:\Users\jasne\Projects
\node\libnode.vcxproj]
          with
          [
              Arg=int &
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(123): message : see reference to function template instantiation 'std::string node::SPrintFImpl<int&,int&,bool&>(const char *,Arg,int &,bool &)' being compiled [C:\Users\jas
ne\Projects\node\libnode.vcxproj]
          with
          [
              Arg=int &
          ]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(128): message : see reference to function template instantiation 'std::string node::SPrintF<int&,int&,bool&>(const char *,int &,int &,bool &)' being compiled [C:\Users\jasne
\Projects\node\libnode.vcxproj]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(137): message : see reference to function template instantiation 'void node::FPrintF<int&,int&,bool&>(FILE *,const char *,int &,int &,bool &)' being compiled [C:\Users\jasne
\Projects\node\libnode.vcxproj]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(150): message : see reference to function template instantiation 'void node::Debug<int32_t&,int32_t&,bool&>(node::EnabledDebugList *,node::DebugCategory,const char *,int32_t
 &,int32_t &,bool &)' being compiled [C:\Users\jasne\Projects\node\libnode.vcxproj]
C:\Users\jasne\Projects\node\src\node_http2.cc(354): message : see reference to function template instantiation 'void node::Debug<int32_t&,int32_t&,bool&>(node::Environment *,node::DebugCategory,const char *,int32_t &,int32_
t &,bool &)' being compiled [C:\Users\jasne\Projects\node\libnode.vcxproj]
C:\Users\jasne\Projects\node\src\debug_utils-inl.h(42,1): warning C4804: '>>=': unsafe use of type 'bool' in operation [C:\Users\jasne\Projects\node\libnode.vcxproj]
```

Reply:
I think that’s just yet another reason to implement https://github.com/nodejs/node/pull/32385#pullrequestreview-378668359. /cc @Himself65

Reply:
Agree. Shouldn't block for this PR tho :-)

Reply:
I think this is cause by implicit conversion in SPrintF 🤔, I will fix it later